### ISSUE TITLE
BadAccess (attempt to access private resource denied) - not starting

### ISSUE THREAD

**Issue by elandorr** [DeaDBeeF-Player/deadbeef#3065]
### Steps to reproduce the problem
Try to start the program.

### What's going on? Describe the problem in as much detail as possible.

It won't start, failing with some X error that a user can't hope to debug:

<details>
  <summary>log</summary>
  
```
<all the included plugins listing>
starting plugin FFMPEG audio player
selected output plugin: PulseAudio output plugin
INFO: from file /home/FOOOO/.config/deadbeef/playlists/0.dbpl
convgui: gtkui plugin not found
shellexecui: can't find gtkui plugin
gtkui plugin compiled for gtk version: 2.16.0
The program 'deadbeef' received an X Window System error.
This probably reflects a bug in the program.
The error was 'BadAccess (attempt to access private resource denied)'.
  (Details: serial 8 error_code 10 request_code 33 minor_code 0)
  (Note to programmers: normally, X errors are reported asynchronously;
   that is, you will receive the error a while after causing it.
   To debug your program, run it with the --sync command line
   option to change this behavior. You can then get a meaningful
   backtrace from your debugger if you break on the gdk_x_error() function.)
```
  
</details>

### Information about the software:

Deadbeef version: latest stable from your website, deadbeef 1.9.6 [static] [portable]
OS: linux 6.8.2 

Every imaginable piece of GTK is installed: https://privatebin.net/?61b593b7734d10cb#WrWtei2xGcV7zzvnNz9xSd6kagTyiMgmuYncb7N94RT

**Comment by Oleksiy-Yakovenko** [comment]
Which OS?

**Comment by Oleksiy-Yakovenko** [comment]
I need to know specific version / build of your installed OS, so that I can install it in a VM and try reproducing your problem.
Just "Debian" would not cut it -- I know for sure that deadbeef works fine on Debian.
Also I need to see the full deadbeef start log, including the plugin list.

**Comment by Oleksiy-Yakovenko** [comment]
One thing that you can try is fully delete ~/.config/deadbeef folder, and delete any non-bundled deadbeef plugins.
Then see if it works after that.
If it does -- it may indicate that the problem was caused either by some plugin, or by using some configuration option.

**Comment by Oleksiy-Yakovenko** [comment]
So in order to trigger the problem, you are removing all these lines from the config file?

**Comment by Oleksiy-Yakovenko** [comment]
I tried doing this, and nothing happened, deadbeef still runs as usual.

**Comment by Oleksiy-Yakovenko** [comment]
So let me confirm whether I understand the repro steps :

1. start deadbeef with empty config
2. quit, and delete those 4 lines from the freshly generated config
3. start deadbeef again, and get a crash

is this correct?
do you even need to delete any lines?

**Comment by Oleksiy-Yakovenko** [comment]
nvm, I think I understand. You mean that when you delete those lines -- deadbeef starts again, not the crash.

**Comment by Oleksiy-Yakovenko** [comment]
It may theoretically be caused by some UI layout configuration, which would produce a weird combination of positions or sizes, which in turn breaks GTK/X11 expectations. Hard to tell without seeing how your UI looks like.

**Comment by Oleksiy-Yakovenko** [comment]
It does look like something broke outside of deadbeef. It's impossible to know what did break without replicating the same system build, and using debugger. You could try following gdb instructions on the wiki, and produce a backtrace. But there's a high chance it would not help, since this is an X11 error, which occurs asynchronously (just as the error message says). So in order to get to the bottom of it - you would need to modify deadbeef code to switch GTK into a synchronous X11 mode, and then MAYBE something could be figured out (but it may also break something else).

**Comment by Oleksiy-Yakovenko** [comment]
It's simple to debug when it's reproducible, but I can't repro -- so I can't debug :)

**Comment by Oleksiy-Yakovenko** [comment]
> Normally you flip a debug log on to get something to work with.

This only works when it's an anticipated error condition, which is handled/logged by the app.
Which is not the situation that you're experiencing.

The issue that you encountered occurs in either GTK or X11 (not in deadbeef code),
and obviously deadbeef code can't anticipate and handle it.

However, it's quite likely that you can find some GTK or X11 logging options to get something to work with.

**Comment by whtyger** [comment]
I observe the same issue on my PC with deadbeef 1.8.8.
Deleting mentioned prefs didn't change anything -- program won't start. Only when I deleted broken profile from ~/.config/, deadbeef started correctly and continues to start after that.
If I return back old profile, the error returns.
As I still have the broken profile, I can help in debugging. Maybe some logging can be activated to compare logs from start with clean profile and broken one?

**Comment by Oleksiy-Yakovenko** [comment]
@whtyger hello,
the logs may be interesting, but not if you're running deadbeef release from 2021.

**Comment by whtyger** [comment]
I see. But if I upgrade, this bug may disappear.
I suppose, it would be a bad idea to try older broken config with the newer program.
Okay, if clearing profile folder solves the issue, it's a viable solution.

**Comment by Oleksiy-Yakovenko** [comment]
If it disappears - it would mean that the bug is already fixed, no?

**Comment by whtyger** [comment]
Well, I upgraded program to 1.10.0, it runs without issues.
But it also upgraded original broken config file to the same version, so we cannot understand now, what was the cause of this issue.

**Comment by Oleksiy-Yakovenko** [comment]
It's not very interesting to me to find a root cause of a bug which is already fixed.
That said, I believe that you may still be able to reproduce it by running older version, even with upgraded config file.

**Comment by whtyger** [comment]
You are right. There's no need to waste efforts which can be put elsewhere.
Workaround is found -- that's good.

**Comment by Oleksiy-Yakovenko** [comment]
Doesn't seem like anything needs / can be done. Closing.
